sonaranalyzer-dotnet/src/SonarAnalyzer.Common/Rules/BooleanLiteralUnnecessaryBase.cs

```diff
@@ -0,0 +1,286 @@
+        }
+
+        protected void CheckForBooleanConstantOnRightReportOnExtendedLocation(TBinaryExpression binaryExpression,
+            TSyntaxKind booleanSyntaxKind, SyntaxNodeAnalysisContext context)
+        {
+            CheckForBooleanConstant(binaryExpression, booleanSyntaxKind, ErrorLocation.Extended, context, leftSide: false);
+        }
+
+        protected void CheckForBooleanConstantOnLeftReportOnInvertedLocation(TBinaryExpression binaryExpression,
+            TSyntaxKind booleanSyntaxKind, SyntaxNodeAnalysisContext context)
+        {
+            CheckForBooleanConstant(binaryExpression, booleanSyntaxKind, ErrorLocation.Inverted, context, leftSide: true);
+        }
+
+        protected void CheckForBooleanConstantOnRightReportOnInvertedLocation(TBinaryExpression binaryExpression,
+            TSyntaxKind booleanSyntaxKind, SyntaxNodeAnalysisContext context)
+        {
+            CheckForBooleanConstant(binaryExpression, booleanSyntaxKind, ErrorLocation.Inverted, context, leftSide: false);
+        }
+
+        protected void CheckForBooleanConstantOnLeftReportOnNormalLocation(TBinaryExpression binaryExpression,
+            TSyntaxKind booleanSyntaxKind, SyntaxNodeAnalysisContext context)
+        {
+            CheckForBooleanConstant(binaryExpression, booleanSyntaxKind, ErrorLocation.Normal, context, leftSide: true);
+        }
+
+        protected void CheckForBooleanConstantOnRightReportOnNormalLocation(TBinaryExpression binaryExpression,
+            TSyntaxKind booleanSyntaxKind, SyntaxNodeAnalysisContext context)
+        {
+            CheckForBooleanConstant(binaryExpression, booleanSyntaxKind, ErrorLocation.Normal, context, leftSide: false);
+        }
+
+        protected void CheckForBooleanConstant(TBinaryExpression binaryExpression, TSyntaxKind booleanSyntaxKind,
+            ErrorLocation errorLocation, SyntaxNodeAnalysisContext context, bool leftSide)
+        {
+            var expression = leftSide ? Left(binaryExpression) : Right(binaryExpression);
+
+            if (!IsKind(expression, booleanSyntaxKind))
+            {
+                return;
+            }
+
+            Location location;
+            switch (errorLocation)
+            {
+                case ErrorLocation.Normal:
+                    location = expression.GetLocation();
+                    break;
+
+                case ErrorLocation.Extended:
+                    location = CalculateExtendedLocation(binaryExpression, leftSide);
+                    break;
+
+                case ErrorLocation.Inverted:
+                    location = CalculateExtendedLocation(binaryExpression, !leftSide);
+                    break;
+
+                default:
+                    location = null;
+                    break;
+            }
+
+            context.ReportDiagnosticWhenActive(Diagnostic.Create(Rule, location));
+        }
+
+        protected Location CalculateExtendedLocation(TBinaryExpression binaryExpression, bool leftSide)
```

Comment:
In C#, it's recommended to prefix boolean (when possible) by `is`, `can`, `has` so I would rename this var into `isLeftSide`.